lib/mongoid/factory.rb

```diff
@@ -23,16 +23,18 @@ module Factory
     #
     # @param [ Class ] klass The class to instantiate from if _type is not present.
     # @param [ Hash ] attributes The document attributes.
+    # @param [ true | false ] defer_callbacks Flag specifies whether callbacks
+    #   should be run.
     #
     # @return [ Document ] The instantiated document.
-    def build(klass, attributes = nil)
+    def build(klass, attributes = nil, defer_callbacks = false)
```

Comment:
If this method and `from_db` are part of public API I'm afraid they would also need to be duplicated.

Reply:
done

Reply:
I tried searching for this on github and while most results are irrelevant the second hit on https://github.com/search?l=Ruby&p=2&q=factory+from_db&type=Code shows someone using this method.

Reply:
IMO if they're not explicitly labeled `@private api` then they are part of the public API. I duplicated the method to add the new args.